Stack/Opc.Ua.Core/Security/Certificates/DirectoryCertificateStore.cs

```diff
@@ -210,6 +210,55 @@ public Task Add(X509Certificate2 certificate, string password = null)
             return Task.CompletedTask;
         }
 
+        /// <inheritdoc/>
+        public Task AddRejected(X509Certificate2 certificate, int maxCertificates)
+        {
```

Comment:
even better might be to accept a X509Certificate2Collection and write/clean up in a single batch

Reply:
Agree